api/resolvers/item.js

```diff
@@ -1120,14 +1125,19 @@ export const createItem = async (parent, { forward, options, ...item }, { me, mo
     item.url = removeTracking(item.url)
   }
 
-  const enforceFee = me ? undefined : (item.parentId ? ANON_COMMENT_FEE : (ANON_POST_FEE + (item.boost || 0)))
+  const { queries: imgQueriesFn, fees: imgFees } = await imageFeesFromText(item.text, { models, me })
+  const enforceFee = (me ? undefined : (item.parentId ? ANON_COMMENT_FEE : (ANON_POST_FEE + (item.boost || 0)))) + imgFees
   item = await serializeInvoicable(
     models.$queryRawUnsafe(
       `${SELECT} FROM create_item($1::JSONB, $2::JSONB, $3::JSONB, '${spamInterval}'::INTERVAL) AS "Item"`,
       JSON.stringify(item), JSON.stringify(fwdUsers), JSON.stringify(options)),
     { models, lnd, hash, hmac, me, enforceFee }
   )
 
+  // TODO run image queries in same transaction as create_item
```

Comment:
Oh, forgot about this TODO. This is important since else, the user might get an error but the item actually was created. Added TODO to PR description.

Reply:
Prisma has interactive txs now: https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions

We can begin doing this but be mindful that each query will require RTT to the db and we are using serializable txs ... interactive txs will make the tx slow to complete and will increase item creation/update failure probability. 

Depending on the nature of the image queries, it might be worth doing them inside `create_item` and `update_item` ... images are kind of fundamental so it seems well worth adding them.

Reply:
> Depending on the nature of the image queries, it might be worth doing them inside create_item and update_item ... images are kind of fundamental so it seems well worth adding them.

I'll do this now. I also put the calculation of image fees into a postgres function now (not pushed yet).